app/amp_start.py

```diff
@@ -22,5 +22,5 @@ class Handler(start.Handler):
 
     def get(self):
         self.env.amp = True
-        self.env.canonical_url = self.get_url('/', lang=self.env.lang)
+        self.env.canonical_url = self.get_url('/', lang=self.params.lang)
```

Comment:
Maybe can you add a short comment explaining why you use self.params.lang here instead of self.env.lang (which should be used in most other cases)?